.github/workflows/python-tests.yml

```diff
@@ -34,37 +34,22 @@ jobs:
       matrix:
         # Add Python '3.11' after released Optuna v3.1
         # since Optuna v3.0.4 depends on scipy<1.9.0 and >=1.7.0
-        python-version: ['3.7', '3.8', '3.9', '3.10']
+        python-version: ['3.7', '3.8', '3.9', '3.10', '3.11']
+        optuna-version: ['optuna>=3.1.0', 'git+https://github.com/optuna/optuna.git']
```

Comment:
I would suggest that there's no need to run unit tests with Optuna's master branch and Python versions 3.7-3.10. Could you skip these cases?